node/test/Infrastructure/Generator.hs

```diff
@@ -31,22 +67,213 @@ genChainUsingModel GeneratorModel{..} =
 -- | Simplified generator model
 --
 -- Small values, simple addresses, and no fees
-simpleModel :: GeneratorModel GivenHash Char
+simpleModel :: GeneratorModel GivenHash Addr
 simpleModel = GeneratorModel {
       gmAllAddresses  = addrs
     , gmEstimateFee   = \_ _ -> 0
-    , gmBoot          = Transaction {
+    , gmBoot          = DSL.Transaction {
                             trFresh = fromIntegral (length addrs) * initBal
                           , trIns   = Set.empty
-                          , trOuts  = [Output a initBal | a <- addrs]
+                          , trOuts  = [DSL.Output a initBal | a <- addrs]
                           , trFee   = 0
                           , trHash  = 0
                           , trExtra = ["Simple bootstrap"]
                           }
     }
   where
-    addrs :: [Char]
-    addrs = ['a' .. 'g']
+    addrs :: [Addr]
+    addrs = Addr <$> [0 .. 7]
 
     initBal :: Value
     initBal = 10000
+
+-- | Instantiate a simple generator
+simpleGen :: Gen (DSL.Chain GivenHash Addr)
+simpleGen = genChainUsingModel simpleModel
+
+-- | Simple instance of a 'DSL.Chain' to an 'Abstract.Chain' translation.
+asAbstractChain
+  :: DSL.Chain GivenHash Addr
+  -> Either IntException (Chain GivenHash Addr, ChainValidity)
+asAbstractChain ch = do
+  addrs <- case chainAddresses ch of
+    []   -> Left  IntEmptyAddresses
+    a:as -> Right (a :| as)
+  params <- simpleParams addrs initTs
+  runIdentity $ translate addrs ch [] params
+  where
+    initTs = fromMaybe [] . safeHead . map toList . toList $ ch
+
+simpleParams
+  :: NonEmpty Addr
+  -> [DSL.Transaction GivenHash Addr]
+  -> Either IntException (Parameters (TransState GivenHash) GivenHash Addr)
+simpleParams addrs initTs = do
+  abstractInitTs <- traverse (dslTransactionToAbstract addrs) initTs
+  return Parameters
+    { slotLeader = mSlotLeader
+    , currentSeed = mCurrentSeed
+    , currentSlot = _tsCurrentSlot
+    , height = mHeight
+    , quality = mQuality
+    , inCommitmentPhase = mInCommitmentPhase
+    , inOpenPhase = mInOpenPhase
+    , inRecoveryPhase = mInRecoveryPhase
+    , maxMempoolSize = 200
+    , k = 2160
+    , initialStakeDistribution = mInitialStakeDistribution addrs
+    , initialSeed = Seed 0
+    , minFee = const 0 -- TODO: QUESTION: what properties should 'minFee' satisfy?
+    , initTransactions = abstractInitTs
+    -- TODO: how do we want to determine the bootstrap stakeholders?
+    , bootstrapStakeholders = Set.fromList $ toList addrs
```

Comment:
Alright, so we should pass an extra parameter to the `simpleParams` function in this case, right?

Reply:
No, I think as an extra parameter in `GeneratorModel`, instantiated in `simpleModel`.

Reply:
But in that case `simpleParams` has no way to accessing the generator model, does it?

Reply:
Oh, I see. Sorry, yeah - I meant adding it to `simpleModel`, but indeed it would have to be passed as an argument to `simpleParams`. I'm easily confused :-(

Reply:
> I'm easily confused :-(

Welcome to the club :)

Ok. I think it is clearer now.